dcpy/models/geospatial/geometry.py

```diff
@@ -0,0 +1,59 @@
+from enum import Enum
+import re
+from typing import TypeAlias
+from pydantic import BaseModel, field_validator
+
+
+class EpsgCode(str, Enum):
+    WGS84 = "EPSG:4326"
+    StatePlane = "EPSG:2263"
+
+
+class StandardGeometryFormat(str, Enum):
+    wkt = "wkt"
+    wkb = "wkb"
+
+
+class PointXYStr(BaseModel, extra="forbid"):
+    """
+    Class with non-wkt representation of point geometry.
+    'point_xy_str' is a pattern which tells the class how to parse this representation.
+    This string must contain both an x and y character once and only once each.
+
+    For example, wkt would be `point_xy_str = "Point(x y)"`
+    Or a column with value "-74.0, 40.7" would be "x y"
+
+    The wkt function of this class assumes that any string to be parsed has x and y
+    each represented by a decimal (digits, followed by period, followed by digits)
+    For now, this does not need to be more flexible, but this could be another optional input
+    """
+
+    point_xy_str: str
+
+    def wkt(self, s: str) -> str:
```

Comment:
Nice addition to the geospatial functionality! Using regex for parsing out a geometry feels a bit dangerous, however. I think it should require some validation from the user side. At a minimum. Perhaps printing a few records asking to match would help.

Reply:
Can you elaborate on "feels a bit dangerous"?

Reply:
Sure, I referred to the possibility of reading in incorrect geospatial data successfully. Say the column of interest is not of a real geospatial data format and may have errors (perhaps it was entered manually). And the worst case where it won't throw a failure, but we end up with wrong data. idk, maybe my fear is invalid.

Reply:
Yeah I get that. But we still have to read in data somehow. I hope that this makes things a little simple twofold
- specifying that the creator of the template file has to choose a specific string format "point_xy_str" rather than provide their own regex, so that there's less room for weirdness and validating the input. In this case, the string provided can only have one "x" and one "y" and needs both. If source data comes in like "70.0, 40.0", I don't think we're going to get much safer in terms of how we want to tell the template how to convert it into a point. It also (currently) won't support multiple formats, so errors will get thrown if things are inconsistent. Which brings me to my second hope of keeping things simple/not dangerous
- This at least will throw errors when the column doesn't match the provided format, rather than try to handle them in some way. We won't catch anything like flipflopped lat and long, but we also don't catch that currently in cases where we have separate lat long columns (like that one facdb dataset...). For this stuff, I hope we can start catching more weirdness with data expectations - whether all rows with geom are in NYC, stuff like that.